File: prog/Workshop10/src/DNAStatsTest.java
import static org.junit.Assert.*;

import org.junit.Test;
import org.junit.BeforeClass;


public class DNAStatsTest {
	
	static DNAStats stats1;
	static DNAStats stats2;
	static DNAStats stats3;
	static String seq1 = "CTAG";
	static String seq2 = "GGGCCC";
	static String seq3 = "GTGGGGGTGATGTCCACGATTACGCCGACCGGCTGG";
	
	@BeforeClass
	public static void setUp() {
		stats1 = DNAStatsFactory.createDNAStats(seq1);
		stats2 = DNAStatsFactory.createDNAStats(seq2);
		stats3 = DNAStatsFactory.createDNAStats(seq3);
	}

	@Test
	public void testGetGCFraction() {
		assertEquals(0.5, stats1.getGCFraction(), 0.001);
	}

}

File: prog/Workshop10/python/dna_stats_test.py
import unittest

from dna_counter import DNACounter

def createDNAStats(seq):
  return DNACounter(seq)

class DNAStatsTest(unittest.TestCase):

  String seq1 = "CTAG"
  String seq2 = "GGGCCC"
  String seq3 = "GTGGGGGTGATGTCCACGATTACGCCGACCGGCTGG"
  DNAStats stats1 = createDNAStats(seq1)
  DNAStats stats2 = createDNAStats(seq2)
  DNAStats stats3 = createDNAStats(seq3)

  def setUp(self):
    """Call before every test case."""
    pass
  
  def tearDown(self):
    """Call after every test case."""
    pass
 
  def testGCFraction(self):
    """Test that the correct fraction of GC base pairs is returned."""
    assert ((stats1.getGCFraction() - 0.25 ) < 0.001